src/AI/AICombatUtils.java

```diff
@@ -54,8 +54,8 @@ public static double scoreAttackAction(Unit unit, GameAction action, GameMap map
     }
     else
     {
-      StrikeParams params = CombatEngine.calculateTerrainDamage(unit,
-          Utils.findShortestPath(unit, action.getMoveLocation(), map), targetLoc, map);
+      GamePath path = new Utils.PathCalcParams(unit, map).findShortestPath(action.getMoveLocation()); // Hmm. TODO: Add unit path getter to GameAction?
```

Comment:
Since you kept the function in `Utils`: why instantiate a `PathCalcParams` instead of just calling the utility function to do it under the hood?

For that matter, is there an argument for putting that logic somewhere else so that you don't have to create an PCP object for the simple case where you just need a single path?

Reply:
I intended to remove it entirely, but you cleverly wrote a bunch of test code I didn't feel like touching.
I guess it *is* a lot simpler at point-of-call to just call a function if you aren't doing anything funky.

There's an argument for almost anything, but the PCP is a nice bundle of params and a big part of this project was getting rid of our blob of overloads.
I don't love putting an extra object on the heap for bookkeeping we don't really "need", but the language makes it hard to avoid doing that without devolving into unlimited overload works.

I've also found I can pass the PCP into `expandSearchNode()` to save a few params.